AsyncDisplayKit/Details/ASRangeController.mm

```diff
@@ -34,6 +34,7 @@ @interface ASRangeController ()
   BOOL _layoutControllerImplementsSetVisibleIndexPaths;
   BOOL _layoutControllerImplementsSetViewportSize;
   NSSet<NSIndexPath *> *_allPreviousIndexPaths;
+  ASWeakSet<ASCellNode *> *_visibleNodes;
```

Comment:
Interesting.  There used to be tracking like this, and when the class was rewritten to be more functional, I forgot about the deletion case :).  I might argue that we should try to handle deletions explicitly, because we could also call the other two range exit methods (although this will incur cost, I do fear that some clients may depend on this for cache invalidation or such).

That said, this is an excellent solution to an important problem, and any further changes would be prospective improvements.  Thanks @Adlai-Holler for working on this challenging issue and the tricky testing that comes with developing such a change!

Should we file an issue for writing a unit test to expose the problem from before this diff?